### ISSUE TITLE
[SECURITY] Insecure default SECRET_KEY check in validate_env allows deployment with known key

### ISSUE THREAD

**Issue by Siddh2024** [Checkora/Checkora#2017]
## Description

In `game/management/commands/validate_env.py` (lines 22-23), the diagnostics script warns about an insecure default key but does not block its use:

```python
if secret_key == 'django-insecure-dev-key-for-local-testing':
    self.stdout.write(self.style.WARNING("   [WARN] SECRET_KEY is using the insecure default value."))
```

## Impact
The script only warns but does not prevent running with the insecure default. If a developer deploys with this default, it exposes the application to session forgery and cryptographic attacks since the key is known to anyone who reads the source code.

## Suggested Fix
Add a hard block — exit with error if the insecure default is detected.

**Comment by Harshitabansal123** [comment]
Hi @Siddh2024,
I'd like to work on this issue under GSSoC'26.
I can update the validation command to fail when the known insecure default `SECRET_KEY` is detected and follow Django's standard error-handling practices while preserving the existing validation flow.
Thank you!

**Comment by Aqsa30nz** [comment]
Hi @Siddh2024,
I'd like to work on this issue under GSSoC'26. My approach would be to update the environment validation command to fail fast when the known insecure default SECRET_KEY is detected, following Django's recommended error-handling practices while keeping the existing validation flow intact. I'll also verify the behavior with appropriate test coverage where applicable.
Could you please assign this issue to me under GSSoC'26? Thank you!
